sklearn/notebooks/hp_tuning_auto_mpg/ml_engine_hp_tuning_training.ipynb

```diff
@@ -0,0 +1,907 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#!/usr/bin/env python\n",
+    "\n",
+    "# Copyright 2018 Google LLC\n",
+    "#\n",
+    "# Licensed under the Apache License, Version 2.0 (the \"License\");\n",
+    "# you may not use this file except in compliance with the License.\n",
+    "# You may obtain a copy of the License at\n",
+    "#\n",
+    "#     http://www.apache.org/licenses/LICENSE-2.0\n",
+    "#\n",
+    "# Unless required by applicable law or agreed to in writing, software\n",
+    "# distributed under the License is distributed on an \"AS IS\" BASIS,\n",
+    "# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
+    "# See the License for the specific language governing permissions and\n",
+    "# limitations under the License."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# scikit-learn HP Tuning on Google Cloud Machine Learning Engine\n",
+    "This notebook trains a model on Google Cloud Machine Learning Engine (ML Engine) using Hyperparameter Tuning to predict a car's Miles Per Gallon. It uses Auto MPG Data Set from [UCI Machine Learning Repository](https://archive.ics.uci.edu/ml/datasets/Auto+MPG).\n",
+    "\n",
+    "Citation: Dua, D. and Karra Taniskidou, E. (2017). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml]. Irvine, CA: University of California, School of Information and Computer Science."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# How to train your model on ML Engine with HP tuning.\n",
+    "Using HP Tuning for training can be done in a few steps:\n",
+    "1. Create your python model file\n",
+    "    1. Add argument parsing for the hyperparameter values. (These values are chosen for you in this notebook)\n",
+    "    1. Add code to download your data from [Google Cloud Storage](https://cloud.google.com/storage) so that ML Engine can use it\n",
+    "    1. Add code to track the performance of your hyperparameter values.\n",
+    "    1. Add code to export and save the model to [Google Cloud Storage](https://cloud.google.com/storage) once ML Engine finishes training the model\n",
+    "1. Prepare a package\n",
+    "1. Submit the training job"
```

Comment:
Would it make sense to use the Python API to submit the job here instead of using gcloud? It is explained here:
https://cloud.google.com/ml-engine/docs/tensorflow/python-guide